espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/web/WebAtomDeserializer.kt

```diff
@@ -0,0 +1,87 @@
+package io.appium.espressoserver.lib.model.web
+
+import com.google.gson.JsonDeserializationContext
+import com.google.gson.JsonDeserializer
+import com.google.gson.JsonElement
+import com.google.gson.JsonParseException
+import io.appium.espressoserver.lib.helpers.AssertHelpers
+import io.appium.espressoserver.lib.helpers.GsonParserHelpers
+import java.lang.reflect.Type
+
+
+// Deserializer for WebAtom
+class WebAtomDeserializer : JsonDeserializer<WebAtom> {
+    @Throws(JsonParseException::class)
+    override fun deserialize(json: JsonElement, paramType: Type?,
+                             paramJsonDeserializationContext: JsonDeserializationContext?): WebAtom {
+
+        if (json.isJsonObject) {
+            val jsonObj = json.asJsonObject
+
+            // Parse the name of the Atom
+            AssertHelpers.assertIsPrimitive(jsonObj, "name");
```

Comment:
AssertHelpers.assertIsPrimitive function could still return the same object if passed

Reply:
It doesn't return anything. It only checks that it's a JSON primitive.

Reply:
I know. I mean we could do it more effectively if this would look like:

` val webAtomName = AssertHelpers.assertIsPrimitive(jsonObj, "name").asString`